src/rpc/mempool.cpp

```diff
@@ -215,6 +219,15 @@ static RPCHelpMan testmempoolaccept()
                         result_inner.pushKV("vsize", virtual_size);
                         UniValue fees(UniValue::VOBJ);
                         fees.pushKV("base", ValueFromAmount(fee));
+                        CHECK_NONFATAL(tx_result.m_result_type == MempoolAcceptResult::ResultType::VALID);
```

Comment:
```suggestion
```

This check looks redundant? are you in an if with the condition `tx_result.m_result_type == MempoolAcceptResult::ResultType::VALID` and `tx_result` is never modified so this confused me while I was reading the code.

I'm missing anything that makes this observation a dum observation?